iModelCore/ECDb/ECDb/SchemaImportContext.cpp

```diff
@@ -403,6 +413,341 @@ bool NoAdditionalForeignKeyConstraintsPolicy::IsException(ECN::NavigationECPrope
         return true;
 
     return m_propertyExceptions.find(navProp.GetId()) != m_propertyExceptions.end();
+}
+
+//***************************************************************************************
+// SqlTypeDetector
+//***************************************************************************************
+
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+const std::vector<std::string> SqlTypeDetector::GetDataTables(DbCR conn) {
+    std::vector<std::string> tables;
+    auto sql = R"sql(
+        SELECT [tbl_name]
+        FROM   [sqlite_master]
+        WHERE   [type] = 'table'
+                AND NOT [tbl_name] LIKE 'ec\_%'     ESCAPE '\'
+                AND NOT [tbl_name] LIKE 'dgn\_%'    ESCAPE '\'
+                AND NOT [tbl_name] LIKE 'be\_%'     ESCAPE '\'
+                AND NOT [tbl_name] LIKE 'sqlite\_%' ESCAPE '\')sql";
+    auto stmt = conn.GetCachedStatement(sql);
+    while (stmt->Step() == BE_SQLITE_ROW) {
+        tables.push_back(stmt->GetValueText(0));
+    }
+    std::sort(tables.begin(), tables.end());
+    return tables;
+}
+
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+const std::vector<std::string> SqlTypeDetector::GetSystemTables(DbCR conn) {
+    std::vector<std::string> tables;
+    auto sql = R"sql(
+        SELECT tbl_name
+        FROM   [sqlite_master]
+        WHERE  [type] = 'table'
+                AND     [tbl_name] LIKE 'ec\_%' ESCAPE '\')sql";
+    auto stmt = conn.GetCachedStatement(sql);
+    while (stmt->Step() == BE_SQLITE_ROW) {
+        tables.push_back(stmt->GetValueText(0));
+    }
+    std::sort(tables.begin(), tables.end());
+    return tables;
+}
+
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+std::string SqlTypeDetector::Join(std::vector<std::string> const& v, const std::string sep) {
+    if (v.empty()) {
+        return "";
     }
+    std::string init = v.front();
+    return std::accumulate(v.begin() + 1, v.end(), init,
+                           [&](std::string& s, const std::string& piece) -> decltype(auto) {
+                               return s.append(sep).append(piece);
+                           });
+}
+
+//---------------------------------------------------------------------------------------
+// @bsimethod
+//---------------------------------------------------------------------------------------
+void SqlTypeDetector::Validate(RE2 const& re) {
+    if (!re.ok()) {
+        LOG.errorv("REGEX: %s\n", re.pattern().c_str());
+        LOG.errorv("ERROR: %s\n", re.error().c_str());
+    }
+    BeAssert(re.ok());

```

Comment:
Why are you both asserting and logging? Asserting means you are confident this will never ever happen, but logging suggests there's a chance it might happen.
(Triggering an assertion causes DEBUG builds to fail or hang, a perpetual annoyance).

Reply:
The check was for only dev time check so i changed them to inline asserts. I have removed the validate method altogether.